app/src/main/java/org/mozilla/focus/autocomplete/UrlAutoCompleteFilter.java

```diff
@@ -20,21 +20,34 @@ public UrlAutoCompleteFilter(Context context) {
         loadUrls(context);
     }
 
+    /**
+     * Our autocomplete list is all lower case, however the search text might be mixed case.
+     * Our autocomplete EditText code does more string comparison, which fails if the suggestion
+     * doesn't exactly match searchText (ie. if casing differs). It's simplest to just build a suggestion
+     * that exactly matches the search text - which is what this method is for:
+     */
+    private static String prepareAutocompleteResult(final String rawSearchText, final String lowerCaseResult) {
+        return rawSearchText + lowerCaseResult.substring(rawSearchText.length());
+    }
+
     @Override
-    public void onFilter(String searchText, InlineAutocompleteEditText view) {
+    public void onFilter(final String rawSearchText, InlineAutocompleteEditText view) {
         if (domains == null || view == null) {
             return;
         }
 
+        // Search terms are all lowercase already, we just need to lowercase the search text
+        final String searchText = rawSearchText.toLowerCase();
```

Comment:
We should specify a locale here. As our domain list is ASCII only it should be good enough to always use Locale.US here.